src/ruby/ext/grpc/extconf.rb

```diff
@@ -24,10 +24,18 @@
 
 ENV['MACOSX_DEPLOYMENT_TARGET'] = '10.7'
 
-ENV['AR'] = RbConfig::CONFIG['AR'] + ' rcs'
-ENV['CC'] = RbConfig::CONFIG['CC']
-ENV['CXX'] = RbConfig::CONFIG['CXX']
-ENV['LD'] = ENV['CC']
+if ENV['AR'].nil?
+    ENV['AR'] = RbConfig::CONFIG['AR'] + ' rcs'
+end
+if ENV['CC'].nil?
```

Comment:
also, add an ` || ENV['CC'].size == 0` after each of these nil checks, to use `RbConfig` defaults on the chance that someone `export CC=''` ?

Reply:
Done.  I am winging it on ruby syntax here so feel free to add corrections.